deployment/ccip/sequence/deploy_ccip.go

```diff
@@ -217,90 +210,59 @@ func deployCCIPSequence(b operations.Bundle, deps operation.TonDeps, in DeployCC
 	if err != nil {
 		return output, err
+	} else if tonContractAddress != nil {
+		output.ReceiverAddress = tonContractAddress
 	}
 
-	// Timelock
-	timelockAddress := deps.CCIPOnChainState[in.ChainSelector].Timelock
-	timelockStorage := timelock.Data{
-		ID:                       in.CCIPConfig.TimelockParams.ID,
-		MinDelay:                 in.CCIPConfig.TimelockParams.MinDelay,
-		Timestamps:               cell.NewDict(256),
-		BlockedFnSelectorsLen:    0,
-		BlockedFnSelectors:       cell.NewDict(32),
-		ExecutorRoleCheckEnabled: true,
-		OpPendingInfo: timelock.OpPendingInfo{
-			ValidAfter:            0,
-			OpFinalizationTimeout: 0,
-			OpPendingID:           big.NewInt(0),
-		},
-		RBAC: rbac.Data{
-			Roles: cell.NewDict(256),
-		},
+	// Invoke deploy Timelock changeset operation
+	deployTimelockInput := mcmsOps.DeployTimelockInput{
+		ContractPath: tonCompiledContracts[state.Timelock].ContractPath,
+		ID:           in.CCIPConfig.TimelockParams.ID,
+		Coins:        tonCompiledContracts[state.Timelock].SuggestedTONCoinsForDeployment,
+		MinDelay:     in.CCIPConfig.TimelockParams.MinDelay,
+		Admin:        in.CCIPConfig.TimelockParams.Admin,
+		Proposers:    in.CCIPConfig.TimelockParams.Proposers,
+		Executors:    in.CCIPConfig.TimelockParams.Executors,
+		Cancellers:   in.CCIPConfig.TimelockParams.Cancellers,
+		Bypassers:    in.CCIPConfig.TimelockParams.Bypassers,
 	}
-	timelockMessageBody := timelock.Init{
-		QueryID:                  0,
-		MinDelay:                 in.CCIPConfig.TimelockParams.MinDelay,
-		Admin:                    in.CCIPConfig.TimelockParams.Admin,
-		Proposers:                common.SnakeRef[common.WrappedAddress](common.WrapAddresses(in.CCIPConfig.TimelockParams.Proposers)),
-		Executors:                common.SnakeRef[common.WrappedAddress](common.WrapAddresses(in.CCIPConfig.TimelockParams.Executors)),
-		Cancellers:               common.SnakeRef[common.WrappedAddress](common.WrapAddresses(in.CCIPConfig.TimelockParams.Cancellers)),
-		Bypassers:                common.SnakeRef[common.WrappedAddress](common.WrapAddresses(in.CCIPConfig.TimelockParams.Bypassers)),
-		ExecutorRoleCheckEnabled: true,
-		OpFinalizationTimeout:    0,
+
+	deployTimelockOutput, err := operations.ExecuteOperation(b, mcmsOps.DeployTimelockOp, deps, deployTimelockInput)
+	if err != nil {
+		return output, err
+	} else if deployTimelockOutput.Output != nil {
+		output.TimelockAddress = newTONContractAddress(*deployTimelockOutput.Output, in.ChainSelector, state.Timelock, in.ContractsSemver, in.ContractsVersionSha)
+	}
+
+	// Invoke deploy MCMS changeset operation
+	deployMCMSInput := mcmsOps.DeployMCMSInput{
+		ContractPath: tonCompiledContracts[state.MCMS].ContractPath,
+		ID:           in.CCIPConfig.MCMSParams.ID,
+		Coins:        tonCompiledContracts[state.MCMS].SuggestedTONCoinsForDeployment,
```

Comment:
For future improvements - MCMS deployment sequence should not be part of CCIP deployment sequence, but a separate sequence that CCIP composes and uses. We need to structure these operations and sequences per protocol and compose them when needed.

MCMS is a completely separate protocol that can be developed, deployed, and managed separately from CCIP - which just depends on it.